### PR TITLE
Move request_controller tests from pre_5_0 branch to develop

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Make request_controller_test executable, move actual tests from pre_5_0 branch, move MockRequest in a separate file, raise cover functions up to 75%.
Related task: [SDLOPEN-583](https://adc.luxoft.com/jira/browse/SDLOPEN-583)

### CODE REVIEW

**Comment by LitvinenkoIra** [comment]
@LuxoftAKutsan please review

File: src/components/application_manager/test/request_controller/request_controller_test.cc [hunk 1]
```diff
@@ -52,91 +55,147 @@ namespace test {
...
-  RequestPtr empty_register_request_;
   RequestControllerSPtr request_ctrl_;
-
-  const uint32_t kDefaultAppHmiLevelNoneRequestsTimeScale = 10;
-  const uint32_t kDefaultAppHmiLevelNoneTimeScaleMaxRequests = 100u;
-  const uint32_t kDefaultAppTimeScaleMaxRequests = 0;
-  const uint32_t kDefaultAppRequestsTimeScale = 0;
-  const uint32_t kDefaultPendingRequestsAmount = 0;
+  RequestPtr empty_mock_request_;
+  const TestSettings kDefaultSettings_;
```
Comment (okozlovlux) [inline src/components/application_manager/test/request_controller/request_controller_test.cc]: @LitvinenkoIra , `kDefaultSettings_` is not global. please rename

File: src/components/application_manager/test/include/application_manager/mock_request.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2016, Ford Motor Company
```
Comment (okozlovlux) [inline src/components/application_manager/test/include/application_manager/mock_request.h]: @LitvinenkoIra , 2017

File: src/components/application_manager/test/request_controller/request_controller_test.cc (hunk 1 above)
Comment (LitvinenkoIra) [inline src/components/application_manager/test/request_controller/request_controller_test.cc]: @okozlovlux Done
9615d669777579d7f87150603520cd6b86250eee

**Comment by LitvinenkoIra** [comment]
Rebuild required

**Comment by LitvinenkoIra** [comment]
Rebuild required

**Comment by LitvinenkoIra** [comment]
Rebuild required

**Pull request merged**